popovers: Fix show three dots when menu is selected

eyuelberga opened the pull request:
Fixes: #23157

After looking at the code for the in-message menus, the reaction button behaves precisely as required, but the three-dot menu is only visible if it has focus. Once you start going through the popover menu items using the keyboard, the 3-dots menu disappears. 

The emoji picker uses a class called `reaction_button_visible`  to achieve the desired behavior, so I used the same technique to fix the menu icons on the sidebar. 
https://github.com/zulip/zulip/blob/4df8c6610f19f08be1564d755aacb6c544eadbbc/static/js/emoji_picker.js#L687

Instead of creating a different function for all the different classes of menu icons (topic, stream, draft..), I used an 'attribute contains' selector to target all of them (they all end with '-sidebar-menu-icon')


For the in-message 3-dots menu, I found a less intrusive way of achieving the same thing. When the menu icon for that message row is triggered the class `has_actions_popover` is added to the element. So, I simply added a style that targets the 3-dots menu on that element.


**Screenshots and screen captures:**

![left-sidebar-preview](https://user-images.githubusercontent.com/30384633/195831123-2d0c8605-5d9c-4687-8e47-be8b01e9ed26.jpg)
![message-feed-preview](https://user-images.githubusercontent.com/30384633/195831129-12050c72-2c9a-4cc8-ac44-4f4e107e7520.jpg)

![sidebar-vid](https://user-images.githubusercontent.com/30384633/195878809-baee7e48-d87d-4538-bd95-8c91371df2c4.gif)

![message-feed-video](https://user-images.githubusercontent.com/30384633/195878040-0f5aa0b5-8614-494e-821b-665532780b9f.gif)




**Self-review checklist**


- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

alya commented:
Thanks! Please clean up your commit history and post again to request a review. https://zulip.readthedocs.io/en/latest/contributing/version-control.html

timabbott commented:
Thanks for debugging this! The engineering approach looks good to me. One thing I notice visually is that the actions popover `\vdots` icons looks faded; I wonder whether we want the more opaque hover CSS while it's being shown as part of displaying the menu?

eyuelberga commented:
> Thanks! Please clean up your commit history and post again to request a review. https://zulip.readthedocs.io/en/latest/contributing/version-control.html

I have cleaned it up. Is it good now?

timabbott commented:
Hmm, maybe we can leave changing the colors for this state to a separate follow-up PR. Can you check whether the other 3-dot menus in the sidebars need similar treatment? There's a lot in the upper-left global filters area; I think the right sidebar doesn't need this because it doesn't have an arrow at all.

The commits are much improved, but the commit messages are very much repeating things that one can see easily by briefly looking at the code itself; it would be much better to spend the commit message body (1) explaining what problem we're solving and (2) talking about why the approach taken is correct -- basically explaining briefly the conclusions of the research you have in the PR description. The line-wrapping also seems like you're only doing 50 characters width in the commit message body body?

eyuelberga commented:
I have updated the commit messages and also adjusted the line wrapping. I have also double-checked that all the menu items in the upper left area are working as expected. 

![global-filters-menu](https://user-images.githubusercontent.com/30384633/196434553-1a28bd62-1152-4c05-86ee-6baa03f925b0.gif)

alya commented:
Looks good to me in manual testing! @timabbott , tagging for your review, since you've been looking at this PR already.

alya commented:
I'm fine with not worrying about the colors until we do a redesign pass; they are also inconsistent with the left sidebar (sigh).

timabbott commented on static/styles/zulip.css:
```diff
@@ -1426,6 +1426,15 @@ td.pointer {
+    visibility: visible !important;
+    pointer-events: all !important;
+    opacity: 1 !important;
+    color: hsl(0, 0%, 73%);
```
Does this work in the dark theme? I'd expect this `color` line to break the dark theme; I would expect just deleting this line to be correct, but haven't tested.

alya replied:
I thought I tested in dark scheme, but I didn't look at the colors super carefully, so I'll leave it to @eyuelberga to comment.

timabbott replied:
We should address this in this PR; having independent changes that we want to think about together is exactly what multiple commits in a single PR is for. I'd remove the line by amending the commit that adds it in this PR, and also add a new 3rd commit that removes the useless line in the similar reaction icon (with its own commit message explaining how you verified that the line had no effect).

eyuelberga replied:
I see. I have now included a third new commit and also updated the previous one.

timabbott replied:
Looks good, thanks!

timabbott commented:
Merged, after adding a bonus commit to remove a focus outline issue that I noticed while testing this. Thanks for making this happen @eyuelberga!
